NOAA-GFDL/FMS
# Bennett & Uriel data_override mini assignment for YAML and data tables

**Description**
Following Uriel's instructions and guidance, made changes to:
- data_override/include/data_override.inc
- test_fms/data_override/test_data_override2.sh
- test_fms/field_manager/test_field_manager2.sh

as well as creating
- test_fms/data_override/test_data_override_init.F90

Fixes # (issue)
Added a namelist named use_data_table_yaml to have default option for user

**How Has This Been Tested?**
Tests labeled in test_fms/data_override/test_data_override2.sh, including:
1. data_override with the default table (i.e not setting the namelist created)
2. data_override with the yaml table (i.e setting the namelist created to .True.)
3. data_override with the default table (i.e setting the namelist created to .True.)

To compile & run (lscamd50-d) inside FMS/build/:
- autoreconf -if ../configure.ac
- module load oneapi mpi compiler hdf5 netcdf libyaml
- set CC=mpiicc
- ../configure --with-yaml FC=`which mpiifort` CC=`which mpiicc` CPPFLAGS="`nc-config --cflags` `nf-config --fflags` -I/opt/libyaml/0.2.5/ONEAPI/2022.2/include -g -traceback" LDFLAGS="-L/opt/libyaml/0.2.5/ONEAPI/2022.2/lib" LIBS="`nc-config --libs` `nf-config --flibs`"
- make check

**Checklist:**
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes
